packages/adapter-next/src/hooks/slice-template-library-read.ts

```diff
@@ -0,0 +1,32 @@
+import type {
+	SliceTemplateLibraryReadHook,
+	SliceTemplateLibraryReadHookData,
+} from "@slicemachine/plugin-kit";
+
+import type { PluginOptions } from "../types";
+
+import * as CallToAction from "../sliceTemplates/CallToAction";
+import { checkIsTypeScriptProject } from "@slicemachine/plugin-kit/fs";
+
+const initialTemplates = [CallToAction];
+
+export const sliceTemplateLibraryRead: SliceTemplateLibraryReadHook<
+	PluginOptions
+> = async (args: SliceTemplateLibraryReadHookData, { helpers }) => {
+	const isTypeScriptProject = await checkIsTypeScriptProject({
+		helpers,
+	});
+
+	const templates =
+		args.templateIds && args.templateIds.length
+			? initialTemplates.filter((t) => args.templateIds?.includes(t.model.id))
```

Comment:
What's the use case for filtering templates here? Shouldn't this just happen on the front end if it needs to?

This adds complexity to the hook which would be better to offload elsewhere to keep the hook dumb (because here it's forcing it to behave quite differently based on options).

Reply:
Basically, it swings the method between:
- If no template IDs return all
- If template IDs return some, but likely just return one (I assume that's your usecase here, only reading one)

Somewhere making it a bit not "pure" 🤔

Reply:
We did it like this because we re-fetch the slice templates list once user has selected templates (they will be able to select more than one day one). But I agree that we could fetch everything once again at creation time and then filter

Reply:
TODO: Discuss that for a change in next iteration